x-pack/plugins/infra/public/components/metrics/sections/chart_section.tsx

```diff
@@ -214,14 +216,17 @@ export const ChartSection = injectI18n(
     }
 
     private handleSelectionBrushEnd = (area: Area) => {
-      const { onChangeRangeTime } = this.props;
+      const { onChangeRangeTime, isLiveStreaming, stopLiveStreaming } = this.props;
       const { startX, endX } = area.domainArea;
       if (onChangeRangeTime) {
         onChangeRangeTime({
           to: endX.valueOf(),
           from: startX.valueOf(),
         } as metricTimeActions.MetricRangeTimeState);
       }
+      if (isLiveStreaming && stopLiveStreaming) {
+        stopLiveStreaming();
+      }
```

Comment:
Could it be worth doing this before calling the `onChangeRangeTime` action in order to minimize the probability for race conditions? :thinking:

Reply:
Sure :+1: